cmds/exp/netbootxyz/network.go

```diff
@@ -14,15 +14,83 @@ import (
 	"net"
 	"net/http"
 	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/u-root/u-root/pkg/dhclient"
 	"github.com/vishvananda/netlink"
 )
 
-func downloadFile(filepath string, url string) error {
+// Counts number of bytes written
+// Conforms to io.Writer interface
+type ProgressCounter struct {
+	Downloaded    uint64
+	Total         uint64
+	PreviousRatio int
+	Writer        io.Writer
+}
+
+func (counter *ProgressCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	counter.Downloaded += uint64(n)
+	counter.PrintProgress()
+	return n, nil
+}
+
+func (counter *ProgressCounter) PrintProgress() {
+	ratio := int(float64(counter.Downloaded) / float64(counter.Total) * 100)
+
+	// Only print every 5% to avoid spamming the serial port and making it look weird
```

Comment:
can the weirdness be fixed? if not too hard, too much work

I would really love continuous progress reporting

Reply:
You can have a look at it and see what it looks like.
I'm not an expert on console interfaces, especially since we have to make it work over serial.

The current implementation prints a line, clears it and prints the next lines.
I found that it causes a lot of flickering and looks not nice.

Unless you have a gigantic image, 5% is continuous enough.
If your image is 2GB then 5% is 100MB. Unless you have a super slow connection, you'll get the 100MB transferred in a few seconds or less.

Reply:
the demo gif looks good, just checking, 5% increment progress is better than nothing -- certainly useful for another case  I was working with.